drivers/usb/device/usb_dc_dw.c

```diff
@@ -814,6 +814,38 @@ int usb_dc_set_address(const u8_t addr)
 	return 0;
 }
 
+int usb_dc_ep_check_cap(const struct usb_dc_ep_cfg_data * const cfg)
+{
+	u8_t ep_idx = USB_DW_EP_ADDR2IDX(cfg->ep_addr);
+
+	SYS_LOG_DBG("ep %x, mps %d, type %d", cfg->ep_addr, cfg->ep_mps,
+		    cfg->ep_type);
+
+	if ((cfg->ep_type == USB_DC_EP_CONTROL) && ep_idx) {
+		SYS_LOG_ERR("invalid endpoint configuration");
+		return -1;
+	}
+
+	if (cfg->ep_mps > QM_USB_MAX_PACKET_SIZE) {
+		SYS_LOG_WRN("unsupported packet size");
+		return -1;
+	}
+
+	if ((USB_DW_EP_ADDR2DIR(cfg->ep_addr) == USB_EP_DIR_OUT) &&
+	    (ep_idx >= QM_USB_OUT_EP_NUM)) {
```

Comment:
I am getting build errors when I try to build with latest master `0824ec6`

`
drivers/usb/device/usb_dc_dw.c: In function ‘usb_dc_ep_check_cap’:
drivers/usb/device/usb_dc_dw.c:829: error: ‘QM_USB_MAX_PACKET_SIZE’ undeclared (first use in this function)
drivers/usb/device/usb_dc_dw.c:829: error: (Each undeclared identifier is reported only once
drivers/usb/device/usb_dc_dw.c:829: error: for each function it appears in.)
drivers/usb/device/usb_dc_dw.c:835: error: ‘QM_USB_OUT_EP_NUM’ undeclared (first use in this function)
drivers/usb/device/usb_dc_dw.c:841: error: ‘QM_USB_IN_EP_NUM’ undeclared (first use in this function)
`

Reply:
Which SoC and board?

Reply:
Intel S1000.
But the point is that the QM_ definitions are all the way in `ext/hal/qmsi/soc/quark_se/include/qm_soc_regs.h`. Why use those definitions in a USB device driver? I had made a suggestion earlier to define one in the USB driver separate from QMSI as the device driver is not only for Quark SoCs

Reply:
I get your point now. Since the driver is used by several platforms I suggest to use device tree for the configuration and capabilities. Also usb_dw_clock_enable() and usb_dw_clock_disable() should also be revised.

Reply:
I agree with using device tree. Will you p;lease address this in a PR?